### PR TITLE
coav-control: Visual Debug

### PR DESCRIPTION
**Pull request by rchiossi**
This patchset integrates vdebug functionality and depth camera visualization into coav-control. It is a compilation of the visualization tool draft implemented my @mbelluzzo and the recent vdebug rework.

This is how it looks like next to gazebo simulation:
ss2

### CODE REVIEW

File: src/detection/DepthImageObstacleDetector.hh [hunk 1]
```diff
@@ -25,8 +25,8 @@
 class DepthImageObstacleDetector : public Detector
 {
 public:
-    DepthImageObstacleDetector(std::shared_ptr<DepthCamera> depth_camera, double threshold_meters = 0.0);
-    const std::vector<Obstacle> &detect() override;
+    DepthImageObstacleDetector(double threshold_meters = 0.0);
+    const std::vector<Obstacle> &detect(std::shared_ptr<void> data) override;
 
 private:
     std::vector<Obstacle> obstacles;
```
Comment (rchiossi) [inline src/detection/DepthImageObstacleDetector.hh]: Indeed. I moved it and didn't noticed I removed it afterwards.

**Comment by rchiossi** [comment]
About your points:
- fair enough
- yeah, copy/paste from the other branch. =(
- Totally agree! I noticed it was better to have a new header way too late. At that point half the commits changed that header, it would be a nightmare to edit them all. If you don't mind, I'll just create the new header in a new commit.

I agree that we should have a better way to inform what data is expected, but I the old way was a cast nightmare. Any suggestion on how to improve it?

The naming issue sounds fair, but I guess even other depth_cameras returning uint16 might need extra fields not provided. Not sure if just changing the name is a good solution here. Maybe a generic data pointer would be better.

**Comment by mbelluzzo** [comment]
"If you don't mind, I'll just create the new header in a new commit.". Sure.

**Comment by rchiossi** [comment]
merged

**Pull request closed without merging**